### DISCUSSION TITLE
how to maintain counter in kong

### DISCUSSION THREAD

**Discussion by kollol-chowdhury-incontact** [Kong/kong#8761] [Help] [answered]
i have a logic that will be implemented when count reaches 100 and then it will be reset to 0.how do i implement it in a thread safe way.

**Comment by ADD-SP** [comment, marked as answer]
I guess what you mean by "thread-safe" is "process-safe", since Nginx is a multi-process model.

You can use `ngx.shared.DICT` provided by OpenResty to share data between multiple processes.

You need to add a configuration in `kong.conf`.

```conf
nginx_http_lua_shared_dict=shdict_counter 5m
```

For usage, please refer to this doc: https://github.com/openresty/lua-nginx-module#ngxshareddict

**Reply by kollol-chowdhury-incontact** [reply]
Thanks for the reply.
Problem is i am trying to aggregate data across all workers and when count reaches 100, should be able to send the data.. But i am not sure if there is any straightforward way to share both count and the data across workers.

**Reply by StarlightIbuki** [reply]
You could implement a simple lock with shm(the access to the shm is atomic). Or, you could use https://github.com/openresty/lua-resty-lock.
